valik kod/lab2.cpp

```diff
@@ -0,0 +1,161 @@
+#include <iostream>
+#include <initializer_list>
+#include <functional>
+#include <memory>
+#include <stdexcept>
+
+template <typename T>
+class LinkedList
+{
+private:
+    struct Node
+    {
+        T data;
+        std::unique_ptr<Node> next;
+        Node(const T &data) : data(data), next(nullptr) {}
+    };
+
+    std::unique_ptr<Node> head;
+    size_t size_;
+
+public:
+    LinkedList() : head(nullptr), size_(0) {}
+
+    LinkedList(std::initializer_list<T> list) : LinkedList()
+    {
+        auto it = list.end();
+        while (it != list.begin())
+        {
+            --it;
+            push_front(*it);
+        }
+    }
+
+    LinkedList(LinkedList &&other) noexcept : head(std::move(other.head)), size_(other.size_)
+    {
+        other.size_ = 0;
+    }
+
+    LinkedList &operator=(LinkedList &&other) noexcept
+    {
+        if (this != &other)
+        {
+            head = std::move(other.head);
+            size_ = other.size_;
+            other.size_ = 0;
+        }
+        return *this;
+    }
+
```

Comment:
Code Simplification: Used for loop for better readability in the constructor that takes std::initializer_list.

Reply:
ok